xarray/plot/__init__.py

```diff
@@ -1,5 +1,6 @@
 from .facetgrid import FacetGrid
 from .plot import contour, contourf, hist, imshow, line, pcolormesh, plot, step
+from .dataset_plot import scatter
```

Comment:
I don't think this is a good idea anymore. The other methods under the plot namespace work only with DataArrays.

Shall we revert this change?

Reply:
Sorry, I don't quite understand what you're trying to say. But please revert whatever you think should be reverted, I don't have strong opinions about these high-level design decisions

Reply:
I think this means that `xr.plot.scatter` will use a dataset plotting method while `xr.plot.imshow` etc. will use DataArray methods. Perhaps better to avoid that?

Reply:
As of now I think that fine, because DataArray doesn't have a scatter method. But I agree that if in the future someone will implement a scatter method it will be confusing.
I'm actually not sure at all why `xr.plot.scatter` should exist. Can we not just remove it?